### PR TITLE
Fix py3 pynio backend

### LINKED ISSUE #1611: pynio backend broken in python 3
pynio recently released a development python 3 port (see https://github.com/NCAR/pynio/issues/10). The pynio backend in xarray needs a slight fix to use this though; the get_variables() method in the NioDataStore class tries to access the iteritems() method of the variables object, which no longer exists in python3. The simple fix is to replace iteritems() with items() in this method. I've made this fix on a local version of the code, and it seems to work as expected. I would be happy to put together a PR if nobody has done such, and if someone could point me to any documentation or recommendations on the preferred conventions for structuring PRs for the project (I have not contributed before). Thanks!

### PR DESCRIPTION
**Pull request by brhillman**
Fix backend for pynio with python 3 support. Also fixes tests for pynio, and adds a test for development version of pynio that includes python 3 support until pynio releases an official version with python 3 support.
 
 - [x] Closes #1611 
 - [x] Tests added / passed
 - [x] Passes ``git diff upstream/master **/*py | flake8 --diff``
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

### CODE REVIEW

**Comment by brhillman** [comment]
I'm not sure why but `git diff upstream/master | flake8 --diff` is reporting some non-conforming lines that seem to either follow conventions used in the surrounding code, and in lines that I did *not* modify (attached).
[flake8_diff.txt](https://github.com/pydata/xarray/files/1360386/flake8_diff.txt)

File: ci/requirements-py36-pynio-dev.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - ncar
+dependencies:
+  - python=3.6
+  - dask
+  - distributed
+  - h5py
+  - h5netcdf
+  - matplotlib
+  - netcdf4
+  - pynio=dev python=3
```
Comment (jhamman) [inline ci/requirements-py36-pynio-dev.yml]: do we need the python=3 piece here?

File: .travis.yml [hunk 2]
```diff
@@ -59,6 +59,8 @@ matrix:
     env: CONDA_ENV=py36-bottleneck-dev
   - python: 3.6
     env: CONDA_ENV=py36-condaforge-rc
+  - python: 3.6
+    env: CONDA_ENV=py36-pynio-dev
```
Comment (jhamman) [inline .travis.yml]: add this same block to just after line 41. Travis requires these `allowed_failures` tests to be listed in the main test matrix too.

File: ci/requirements-py36-pynio-dev.yml (hunk 1 above)
Comment (brhillman) [reply ci/requirements-py36-pynio-dev.yml]: Nope, fixed.

File: .travis.yml (hunk 2 above)
Comment (brhillman) [reply .travis.yml]: Gotcha. Done.

**Comment by jhamman** [comment]
This looks like its complete. Thanks @brhillman!

@shoyer - any problem adding another dev test for pynio? I think this should be a temporary addition, one that can be removed after pynio produces a full release with python3 support.

**Comment by brhillman** [comment]
@jhamman no problem, more than happy to finally be able to contribute in some way. Thanks for your help putting this together!

**Pull request merged**